instance/models/server.py

```diff
@@ -287,8 +289,8 @@ def public_ip(self):
 
         try:
             public_addr = openstack.get_server_public_address(self.os_server)
-        except novaclient.exceptions.NotFound:
-            return None  # This server has been deleted.
+        except Exception:  # pylint: disable=broad-except
```

Comment:
I'd rather have a specific list of exceptions or exception categories listed, to only catch things that come from the API itself - broad excepts are truly evil, here it could hide exceptions thrown by our code.